### PR TITLE
Kotlin/schedule

### PR DESCRIPTION
**Pull request by fourlastor**
This moves the Schedule service implementation to kotlin, hurray!

### CODE REVIEW

File: app/src/main/java/net/squanchy/schedule/ScheduleService.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            eventsObservable
+                    .withLatestFrom(daysObservable, combineEventsById())
+                    .subscribeOn(Schedulers.io())
+        }
+    }
+
+    private fun combineEventsById(): BiFunction<Map<String, List<Event>>, FirebaseDays, Schedule> {
+        return BiFunction { eventsMap, (days) ->
+            val pages = eventsMap.keys
+                    .map { key ->
```
Comment (tiwiz) [inline app/src/main/java/net/squanchy/schedule/ScheduleService.kt]: Can you split this into a separate function? It looks a bit hard to understand to me

File: app/src/main/java/net/squanchy/schedule/ScheduleService.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                        days!!.find {
+                            (id) ->
+                            id == key
+                        }
+                    }
+                    .filterNotNull()
+                    .map {
+                        val date = LocalDateTime.parse(it.date!!, DATE_FORMATTER)
```
Comment (tiwiz) [inline app/src/main/java/net/squanchy/schedule/ScheduleService.kt]: Same here :-)

**Review by tiwiz — CHANGES_REQUESTED** [review]

File: app/src/main/java/net/squanchy/schedule/ScheduleService.kt (hunk 2 above)
Comment (fourlastor) [reply app/src/main/java/net/squanchy/schedule/ScheduleService.kt]: 6c5da1e

File: app/src/main/java/net/squanchy/schedule/ScheduleService.kt (hunk 1 above)
Comment (fourlastor) [reply app/src/main/java/net/squanchy/schedule/ScheduleService.kt]: 6c5da1e

**Comment by fourlastor** [comment]
Putting an :x: because CI is missing

**Pull request merged**